proxy/src/console/provider.rs

```diff
@@ -250,18 +252,19 @@ pub struct NodeInfo {
 
 pub type NodeInfoCache = TimedLru<Arc<str>, NodeInfo>;
 pub type CachedNodeInfo = timed_lru::Cached<&'static NodeInfoCache>;
-pub type AllowedIpsCache = TimedLru<Arc<str>, Arc<Vec<String>>>;
+pub type AllowedIpsCache = TimedLru<SmolStr, Arc<Vec<String>>>;
+pub type RoleSecretCache = TimedLru<(SmolStr, SmolStr), Option<AuthSecret>>;
```

Comment:
What's the reasoning here for having these be separate caches? I would have thought they be combined

Reply:
For the role secret it depends on a role and for ips not.

Also, for the following change with redis it would be better to manage if they are separate (than we can invalidate ips on update without invalidating all secrets for this ep and wise versa).

But honestly, I am still not sure, which approach is better.

Reply:
That makes sense to me. Thanks